### PR TITLE
Update docs after the change of retry logic in Installer

### PR DESCRIPTION
**Pull request by kubadz**
**Description**

Changes proposed in this pull request:

- update documentation for retries

**Related issue(s)**

#7055

### CODE REVIEW

File: docs/kyma/04-12-details.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+type: Installation
+---
+
+This section describes the installation mechanisms that may be useful to know when installing, updating or upgrading Kyma.
+
+### Retry policy
+
+Kyma Operator has a retry mechanism which makes sure that the installation will not fail in case of a temporary issues like unavailability of Tiller or a prolonged process of creating a resource path for CustomResources in Kubernetes API Server. 
+
+If the problem occurs during the installation of a one of defined components, the installation of that particular component is retried a few times with an increasing time between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting `action: install` label on the Installation CR.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/kyma/04-12-details.md]:
```suggestion
If the problem occurs during the installation of a one of defined components, the installation of that particular component is retried a few times with a configurable time between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting `action: install` label on the Installation CR. 
```

File: docs/kyma/04-12-details.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+This section describes the installation mechanisms that may be useful to know when installing, updating or upgrading Kyma.
+
+### Retry policy
+
+Kyma Operator has a retry mechanism which makes sure that the installation will not fail in case of a temporary issues like unavailability of Tiller or a prolonged process of creating a resource path for CustomResources in Kubernetes API Server. 
+
+If the problem occurs during the installation of a one of defined components, the installation of that particular component is retried a few times with an increasing time between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting `action: install` label on the Installation CR. 
+
+> **NOTE:** The configuration of retries can be adjusted by setting `backoffIntervals` argument in Installer Deployment. The value is a comma-separated list of seconds to wait between consecutive retries. For example, the default one is: `10,20,40,60,80`.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/kyma/04-12-details.md]:
```suggestion
> **NOTE:** The configuration of retries can be adjusted by setting `backoffIntervals` argument in Installer Deployment. The value is a comma-separated list of seconds to wait between consecutive retries. It also defines the total number of retries. For example, the default one is: `10,20,40,60,80`.
```

File: docs/kyma/04-12-details.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+title: Details
+type: Installation
+---
+
+This section describes in details some of the installation mechanisms.
+
+### Retry policy
+
+Kyma Operator has a retry mechanism which makes sure that the installation does not fail in case of temporary issues like unavailability of Tiller or a prolonged process of creating a resource path for CustomResources in Kubernetes API Server.
```
Comment (majakurcius) [inline docs/kyma/04-12-details.md]:
```suggestion
Kyma Operator has a retry mechanism which makes sure that the installation does not fail in the case of temporary issues, such as Tiller unavailability or a prolonged process of creating a resource path for custom resources in the Kubernetes API Server. 
```

File: docs/kyma/04-12-details.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+type: Installation
+---
+
+This section describes in details some of the installation mechanisms.
+
+### Retry policy
+
+Kyma Operator has a retry mechanism which makes sure that the installation does not fail in case of temporary issues like unavailability of Tiller or a prolonged process of creating a resource path for CustomResources in Kubernetes API Server. 
+
+If an error occurs during the installation of a component, the installation of that particular component is retried based on the provided configuration that specifies intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting `action: install` label on the Installation CR.
```
Comment (majakurcius) [inline docs/kyma/04-12-details.md]:
```suggestion
If an error occurs during the installation of a component, the installation of that particular component is retried based on the provided configuration that specifies the intervals between consecutive attempts. The default configuration consists of 5 retries with the following time intervals between each attempt: 10s, 20s, 40s, 60s, 80s. After that, the installation is stopped and can be restarted manually by setting the `action: install` label on the Installation CR.
```

File: docs/kyma/04-12-details.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+> **NOTE:** The configuration of retries can be adjusted by setting `backoffIntervals` argument in Installer Deployment. The value is a comma-separated list of numbers that represent intervals between consecutive retries. It also defines the total number of retries. For example, the default one is: `10,20,40,60,80`.
```
Comment (majakurcius) [inline docs/kyma/04-12-details.md]:
```suggestion
> **NOTE:** The configuration of retries can be adjusted by setting the `backoffIntervals` argument in the Installer Deployment. The value is a comma-separated list of numbers that represent the intervals between consecutive retries. It also defines the total number of retries. For example, the default one is: `10,20,40,60,80`.
```

File: docs/kyma/04-12-details.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+In case of an error occurring before the installation of components, the installation will be retried every 30 seconds until successful.
```
Comment (majakurcius) [inline docs/kyma/04-12-details.md]:
```suggestion
In the case of an error occurring before the installation of components, the installation is retried every 30 seconds until successful. 
```

File: docs/kyma/05-04-component-source-url.md [hunk 7]
```diff
@@ -68,6 +68,6 @@ To specify the exact location of the `Chart.yaml`, append it to the URL beginnin
 
 ### Error handling and retry policy
 
-If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it three times during the installation process. If it fails to reach the specified URL in one of the three attempts or fails to find the required files, the installation step fails and the entire installation process starts over.
+If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it three times during the installation process. If it fails to reach the specified URL in one of the three attempts or fails to find the required files, the installation step fails and the process of installing the component is repeated according to the default installation retry process.
```
Comment (majakurcius) [inline docs/kyma/05-04-component-source-url.md]:
```suggestion
If you specify an external URL as a source for a Kyma component, the Kyma Operator attempts to access it three times during the installation process. If it fails to reach the specified URL in one of the three attempts, or fails to find the required files, the installation step fails and the component installation is repeated according to the default installation retry process.
```

File: docs/kyma/06-01-installation.md [hunk 8]
```diff
@@ -77,8 +77,8 @@ The **status.state** field uses one of the following four values to describe the
 |----------|-------------|
 | **Installed** | Installation successful. |
 | **Uninstalled** | Uninstallation successful. |
-| **InProgress** | The Installer is still installing or uninstalling Kyma. No errors logged. |
-| **Error** | The Installer encountered a problem but it continues to try to process the resource. |
+| **InProgress** | The Installer is still installing or uninstalling Kyma. No errors for the current installation step are logged. |
```
Comment (majakurcius) [inline docs/kyma/06-01-installation.md]:
```suggestion
| **InProgress** | The Installer is still installing or uninstalling Kyma. No errors for the current installation step are logged. |
```

Comment (kubadz) [reply docs/kyma/06-01-installation.md]: IMHO it is important to keep the information, that no errors have been logged for `the current installation step`. What do you think about:
`The process of (un)installing Kyma is running and no errors for the current installation step have been logged.`
?

Comment (majakurcius) [reply docs/kyma/06-01-installation.md]: I also wondered about that, but I figured if it was important you'd point that out anyway, and voila :D 
That sounds good, let's go with that, just skip the "installation" in `installation step` :)

**Pull request merged**